mitodl/open-discussions
# 4247: Migrate Django users to Keycloak

#### Pre-Flight checklist

#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/4247

#### What's this PR do?

1. Adds a new Django app, `keycloak_user_export` to the Open Discussions project.
2. The new app includes a table called UserExportToKeycloak which has a foreign key relationship with the Django User model.
3. A management task, `migrate_users_to_keycloak`, is included which attempts to create a Keycloak user for each Django user who has a password defined and does not have a social-auth record associated with the ol-oidc provider.

#### How should this be manually tested?
To perform the test, you should have:
1. Keycloak running locally.
2. Open Discussions running locally.
3. A few users created in Open Discussions.

1. Using this branch, run `migrate` to create the `UserExportToKeycloak` table.  The records for this table should be visible in Django Admin.
4. Run the management command via: `open-discussions % docker-compose run web ./manage.py migrate_users_to_keycloak <KEYCLOAK_ADMIN_EMAIL> <KEYCLOAK_ADMIN_PASSWORD> admin-cli <ADMIN_CLI_CLIENT_SECRET>`
5. Verify that new UserExportToKeycloak records are created.
6. Verify that you can login to Keycloak using the email and password associated with the Django users.
